=== modified file 'includes/inc_request.php'
--- includes/inc_request.php	2013-10-22 13:00:22 +0000
+++ includes/inc_request.php	2014-01-26 19:30:55 +0000
@@ -8,7 +8,7 @@
 	private $rootpath = '../';

 	private $parameters = array();
-	private $render_keys = array('type', 'value', 'size', 'maxlength', 'style', 'label', 'checked', 'onchange', 'autocomplete');
+	private $render_keys = array('type', 'value', 'size', 'maxlength', 'style', 'label', 'checked', 'onchange', 'onkeyup', 'autocomplete');
 	private $validation_keys = array('not_empty', 'match');

 	private $output = 'tr';

=== modified file 'transactions/many_to_one.php'
--- transactions/many_to_one.php	2013-10-22 13:00:22 +0000
+++ transactions/many_to_one.php	2014-01-26 16:55:17 +0000
@@ -19,7 +19,9 @@
 //status 7: extern

 $req = new request('admin');
-$req->add('fixed', 10, 'post', array('type' => 'text', 'size' => 3, 'maxlength' => 3, 'label' => 'Vast bedrag'), array('match' => 'positive'))
+$req->add('fixed', 0, 'post', array('type' => 'text', 'size' => 4, 'maxlength' => 4, 'label' => 'Vast bedrag'), array('match' => 'positive'))
+	->add('percentage', 0, 'post', array('type' => 'text', 'size' => 4, 'maxlength' => 4, 'label' => 'Percentage'))
+	->add('percentage_base', 0, 'post', array('type' => 'text', 'size' => 4, 'maxlength' => 4, 'label' => 'Percentage saldo-basis'))
 	->add('fill_in', '', 'post', array('type' => 'submit', 'label' => 'Vul in'))
 	->add('no_newcomers', '', 'post', array('type' => 'checkbox', 'label' => 'Geen instappers.'), array())
 	->add('no_leavers', '', 'post', array('type' => 'checkbox', 'label' => 'Geen uitstappers.'), array())
@@ -37,7 +39,7 @@
 $to_user_id = null;

 foreach($active_users as $user){
-	$req->add('amount-'.$user['id'], 0, 'post', array('type' => 'text', 'size' => 3, 'maxlength' => 3, 'onchange' => 'recalc_table_sum(this);'), array('match' => 'positive'));
+	$req->add('amount-'.$user['id'], 0, 'post', array('type' => 'text', 'size' => 3, 'maxlength' => 3, 'onkeyup' => 'recalc_table_sum(this);'), array('match' => 'positive'));
 	if ($letscode_to && $user['letscode'] == $letscode_to){
 		$to_user_id = $user['id'];
 		$to_user_fullname = $user['fullname'];
@@ -79,8 +81,11 @@
 }

 $fixed = $req->get('fixed');
+$percentage = $req->get('percentage');
+$percentage_base = $req->get('percentage_base');
+$perc = 0;

-if ($req->get('fill_in') && $fixed){
+if ($req->get('fill_in') && ($fixed || $percentage)){
 	foreach ($active_users as $user){
 		if ($user['letscode'] == $req->get('letscode_to')
 			|| (check_newcomer($user['adate']) && $req->get('no_newcomers'))
@@ -88,7 +93,11 @@
 			|| ($user['saldo'] < $user['minlimit'] && $req->get('no_min_limit'))){
 			$req->set('amount-'.$user['id'], 0);
 		} else {
-			$req->set('amount-'.$user['id'], $fixed);
+			if ($percentage){
+				$perc = round(($user['saldo'] - $percentage_base)*$percentage/100);
+				$perc = ($perc > 0) ? $perc : 0;
+			}
+			$req->set('amount-'.$user['id'], $fixed + $perc);
 		}
 	}
 }
@@ -128,11 +137,12 @@

 function show_form($req, $data_table){
 	echo '<form method="post"><div style="background-color:#ffdddd; padding:10px;">';
-	echo '<p><strong>Een vast bedrag invullen voor alle rekeningen.</strong></p>';
+	echo '<p><strong>Een vast bedrag en/of percentage invullen voor alle rekeningen.</strong></p>';
 	echo '<table  cellspacing="5" cellpadding="0" border="0">';
-	$req->set_output('tr')->render(array('fixed', 'no_newcomers', 'no_leavers', 'no_min_limit', 'fill_in'));
+	$req->set_output('tr')->render(array('fixed', 'percentage', 'percentage_base', 'no_newcomers', 'no_leavers', 'no_min_limit', 'fill_in'));
 	echo '</table>';
-	echo '<p><strong><i>Aan LETSCode</i></strong> wordt altijd automatisch overgeslagen. Alle bedragen blijven individueel aanpasbaar alvorens de massa-transactie uitgevoerd wordt.</p>';
+	echo '<p><strong><i>Aan LETSCode</i></strong> wordt altijd automatisch overgeslagen. Alle bedragen blijven individueel aanpasbaar alvorens de massa-transactie uitgevoerd wordt.</p>';
+	echo '<p><strong><i>Je kan een vast bedrag en/of een percentage op het saldo invullen</i></strong> Als een percentage wordt ingevuld, worden de bedragen berekend t.o.v. percentage saldo basis.</p>';
 	echo '</div><br/>';
 	echo '<div id="transformdiv" style="padding:10px;">';
 	$data_table->render();

=== modified file 'transactions/one_to_many.php'
--- transactions/one_to_many.php	2013-10-22 13:00:22 +0000
+++ transactions/one_to_many.php	2014-01-26 19:29:22 +0000
@@ -18,7 +18,9 @@
 //status 7: extern

 $req = new request('admin');
-$req->add('fixed', 10, 'post', array('type' => 'text', 'size' => 3, 'maxlength' => 3, 'label' => 'Vast bedrag'), array('match' => 'positive'))
+$req->add('fixed', 0, 'post', array('type' => 'text', 'size' => 3, 'maxlength' => 3, 'label' => 'Vast bedrag'), array('match' => 'positive'))
+	->add('percentage', 0, 'post', array('type' => 'text', 'size' => 4, 'maxlength' => 4, 'label' => 'Percentage'))
+	->add('percentage_base', 0, 'post', array('type' => 'text', 'size' => 4, 'maxlength' => 4, 'label' => 'Percentage saldo-basis'))
 	->add('fill_in', '', 'post', array('type' => 'submit', 'label' => 'Vul in'))
 	->add('no_newcomers', '', 'post', array('type' => 'checkbox', 'label' => 'Geen instappers.'), array())
 	->add('no_leavers', '', 'post', array('type' => 'checkbox', 'label' => 'Geen uitstappers.'), array())
@@ -36,7 +38,7 @@
 $from_user_id = null;

 foreach($active_users as $user){
-	$req->add('amount-'.$user['id'], 0, 'post', array('type' => 'text', 'size' => 3, 'maxlength' => 3, 'onchange' => 'recalc_table_sum(this);'), array('match' => 'positive'));
+	$req->add('amount-'.$user['id'], 0, 'post', array('type' => 'text', 'size' => 3, 'maxlength' => 3, 'onkeyup' => 'recalc_table_sum(this);', 'onchange' => 'recalc_table_sum(this)'), array('match' => 'positive'));
 	if ($letscode_from && $user['letscode'] == $letscode_from){
 		$from_user_id = $user['id'];
 		$from_user_fullname = $user['fullname'];
@@ -78,8 +80,11 @@
 }

 $fixed = $req->get('fixed');
+$percentage = $req->get('percentage');
+$percentage_base = $req->get('percentage_base');
+$perc = 0;

-if ($req->get('fill_in') && $fixed){
+if ($req->get('fill_in') && ($fixed || $percentage)){
 	foreach ($active_users as $user){
 		if ($user['letscode'] == $req->get('letscode_from')
 			|| (check_newcomer($user['adate']) && $req->get('no_newcomers'))
@@ -87,7 +92,11 @@
 			|| ($user['saldo'] > $user['maxlimit'] && $req->get('no_max_limit'))){
 			$req->set('amount-'.$user['id'], 0);
 		} else {
-			$req->set('amount-'.$user['id'], $fixed);
+			if ($percentage){
+				$perc = round(($user['saldo'] - $percentage_base)*$percentage/100);
+				$perc = ($perc > 0) ? $perc : 0;
+			}
+			$req->set('amount-'.$user['id'], $fixed + $perc);
 		}
 	}
 }
@@ -135,11 +144,12 @@
 	$req->set_output('tr')->render(array('confirm_password', 'zend', 'transid'));
 	echo '</table></div>';
 	echo '<div style="background-color:#ffdddd; padding: 10px;">';
-	echo '<p><strong>Een vast bedrag invullen voor alle rekeningen.</strong></p>';
+	echo '<p><strong>Een vast bedrag en/of percentage invullen voor alle rekeningen.</strong></p>';
 	echo '<table  cellspacing="5" cellpadding="0" border="0">';
-	$req->set_output('tr')->render(array('fixed', 'no_newcomers', 'no_leavers', 'no_max_limit', 'fill_in', 'transid'));
+	$req->set_output('tr')->render(array('fixed', 'percentage', 'percentage_base', 'no_newcomers', 'no_leavers', 'no_max_limit', 'fill_in', 'transid'));
 	echo '</table>';
-	echo '<p><strong><i>Van LETSCode</i></strong> wordt altijd automatisch overgeslagen. Alle bedragen blijven individueel aanpasbaar alvorens de massa-transactie uitgevoerd wordt.</p>';
+	echo '<p><strong><i>Van LETSCode</i></strong> wordt altijd automatisch overgeslagen. Alle bedragen blijven individueel aanpasbaar alvorens de massa-transactie uitgevoerd wordt.</p>';
+	echo '<p><strong><i>Je kan een vast bedrag en/of een percentage op het saldo invullen</i></strong> Als een percentage wordt ingevuld, worden de bedragen berekend t.o.v. percentage saldo basis.</p>';
 	echo '</div><br/><table>';
 	$req->set_output('tr')->render('refresh');
 	echo '</table></form>';
